scripts/scil_compute_connectivity.py

```diff
@@ -0,0 +1,382 @@
+
+Current strategy is to keep the longest streamline segment connecting
+2 regions. If the streamline crosses other gray matter regions before
+reaching its final connected region, the kept connection is still the
+longest.
+
+This is robust to compressed streamlines.
+
+NOTE: this script can take a while to run. Please be patient.
+"""
+
+from __future__ import division
+
+from builtins import zip
+import argparse
+import logging
+import os
+import time
+
+from dipy.tracking.streamlinespeed import length
+import nibabel as nb
+import numpy as np
+
+from scilpy.io.streamlines import (load_trk_in_voxel_space,
+                                   save_from_voxel_space)
+from scilpy.io.utils import (add_overwrite_arg,
+                             assert_inputs_exist,
+                             assert_output_dirs_exist_and_empty)
+from scilpy.tractanalysis.features import (remove_outliers,
+                                           remove_loops_and_sharp_turns)
+from scilpy.tractanalysis.tools import (compute_connectivity,
+                                        compute_streamline_segment,
+                                        extract_longest_segments_from_profile)
+from scilpy.tractanalysis.uncompress import uncompress
+
+
+def _get_output_paths(root_dir):
+    paths = {'raw': os.path.join(root_dir, 'raw_connections/'),
+             'final': os.path.join(root_dir, 'final_connections/'),
+             'removed_length': os.path.join(root_dir, 'removed_length/'),
+             'loops': os.path.join(root_dir, 'loops/'),
+             'outliers': os.path.join(root_dir, 'outliers/'),
+             'qb_loops': os.path.join(root_dir, 'qb_loops/'),
+             'pruned': os.path.join(root_dir, 'pruned/'),
+             'no_loops': os.path.join(root_dir, 'no_loops/'),
+             'no_outliers': os.path.join(root_dir, 'no_outliers/')}
+
+    return paths
+
+
+def _get_saving_options(args):
+    saving_options = {'raw': args.save_raw_connections,
+                      'intermediate': args.save_intermediate,
+                      'discarded': args.save_discarded,
+                      'final': True}
+
+    return saving_options
+
+
+def _create_required_output_dirs(out_paths, args):
+    os.mkdir(out_paths['final'])
+
+    if args.save_raw_connections:
+        os.mkdir(out_paths['raw'])
+
+    if args.save_discarded:
+        os.mkdir(out_paths['loops'])
+        os.mkdir(out_paths['outliers'])
+        os.mkdir(out_paths['qb_loops'])
+        os.mkdir(out_paths['removed_length'])
+
+    if args.save_intermediate:
+        os.mkdir(out_paths['pruned'])
+        os.mkdir(out_paths['no_loops'])
+        os.mkdir(out_paths['no_outliers'])
+
+
+def _save_if_needed(streamlines, args, saving_options, out_paths,
+                    save_type, step_type, in_label, out_label):
+    if saving_options[save_type] and len(streamlines):
+        save_from_voxel_space(streamlines, args.labels, args.tracts,
```

Comment:
"args.tracts" should be changed to : "args.tracks"
```suggestion
        save_from_voxel_space(streamlines, args.labels, args.tracks,
```